opts.go

```diff
@@ -21,11 +21,24 @@ import (
 	"strings"
 
 	cnilibrary "github.com/containernetworking/cni/libcni"
+	cnitypes "github.com/containernetworking/cni/pkg/types/current"
 	"github.com/pkg/errors"
 )
 
 type CNIOpt func(c *libcni) error
 
+func checkVersion(conf *cnilibrary.NetworkConfig) bool {
+	if (conf.Network.CNIVersion == "") {
+		return (conf.Network.Type != "loopback")
+	}
+	for _, version := range cnitypes.SupportedVersions {
```

Comment:
I like the idea of checking supported versions and the idea of the other checks below..